ext/snmp/snmp.c

```diff
@@ -953,7 +953,11 @@ static int netsnmp_session_set_sec_protocol(struct snmp_session *s, char *prot)
 		s->securityPrivProtoLen = USM_PRIV_PROTO_AES_LEN;
 #endif
 	} else {
-		php_error_docref(NULL, E_WARNING, "Unknown security protocol '%s'", prot);
+		zend_value_error("Security protocol must be DES"
+#ifdef HAVE_AES
+			", AES128, or AES"
```

Comment:
Pretty clever solution :D But I think putting the whole `zend_value_error()` into an ifdef and an `else` branch would be more clean and would allow us to use `either` or `one of`.